lib/modules/dosomething/dosomething_kudos/dosomething_kudos.module

```diff
@@ -245,7 +245,10 @@ function dosomething_kudos_get_kudos_for_user_by_file($fid) {
 function dosomething_kudos_term_is_selected($reportback_item, $term) {
   // We need to check both array & objects b/c preprocess_node works w/ RB Item objects
   // and preprocess_user works w/ content arrays.
-  if ((is_array($reportback_item) && empty($reportback_item['existing_kids'])) && ! $reportback_item->existing_kids) {
+  if (is_array($reportback_item) && empty($reportback_item['existing_kids'])) {
+    return false;
+  }
+  else if (!is_array($reportback_item) && !$reportback_item->existing_kids) {
```

Comment:
Just a thought, but maybe we could cast `$reportback_item` to an object here and then we can just avoid the logic of having to check if it's an array it has the item, or if it's a object and it has the property. Then we always just check one thing and keep the logic clean and concise.

```php
$reportback_item = (object) $reportback_item;

if (! $reportback_item->existing_kids) {
  return false;
}
```

thoughts @joe @angaither ?